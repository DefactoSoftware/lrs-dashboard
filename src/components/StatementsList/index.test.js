import React from 'react';
import StatementList from './index.js';
import renderer from 'react-test-renderer';

it('renders no statements when no statements are present', () => {
  const statements = [];

  const tree = renderer.create(<StatementList statements={statements} />).toJSON();

  expect(tree).toMatchSnapshot();
});


it('renders three statements when three statements are present', () => {
  const statements = [
    {
      id: "1234",
      actor: 'foo@bar.com',
      verb: 'created',
      object: 'Chapter',
    },
    {
      id: "5768",
      actor: 'bar@foo.com',
      verb: 'updated',
      object: 'Space',
    },
    {
      id: "9012",
      actor: 'foo@bar.com',
      verb: 'created',
      object: 'Portrait',
    },
  ];

  const tree = renderer.create(<StatementList statements={statements} />).toJSON();

  expect(tree).toMatchSnapshot();
});
