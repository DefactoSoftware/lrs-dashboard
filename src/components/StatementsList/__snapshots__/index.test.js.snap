exports[`test renders no statements when no statements are present 1`] = `
<ul
  className="StatementsList" />
`;

exports[`test renders three statements when three statements are present 1`] = `
<ul
  className="StatementsList">
  <li
    className="item">
    <p>
      <span>
        foo@bar.com
      </span>
       
      <strong>
        created
      </strong>
       
      <span>
        Chapter
      </span>
    </p>
  </li>
  <li
    className="item">
    <p>
      <span>
        bar@foo.com
      </span>
       
      <strong>
        updated
      </strong>
       
      <span>
        Space
      </span>
    </p>
  </li>
  <li
    className="item">
    <p>
      <span>
        foo@bar.com
      </span>
       
      <strong>
        created
      </strong>
       
      <span>
        Portrait
      </span>
    </p>
  </li>
</ul>
`;
